### PR TITLE
[FIX] JMAP: Set configurable limits for /get /set objects

### PR DESCRIPTION
**Pull request by chibenwa**
Help on fixup welcome

### CODE REVIEW

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/core/JmapRfc8621Configuration.scala [hunk 1]
```diff
@@ -56,6 +58,9 @@ object JmapRfc8621Configuration {
   val MAX_SIZE_ATTACHMENTS_PER_MAIL_DEFAULT: MaxSizeAttachmentsPerEmail = MaxSizeAttachmentsPerEmail.of(Size.of(20_000_000L, Size.Unit.B)).get
   val JMAP_UPLOAD_QUOTA_LIMIT_DEFAULT: JmapUploadQuotaLimit = JmapUploadQuotaLimit.of(Size.of(200L, Size.Unit.M)).get
   val JMAP_EMAIL_GET_FULL_MAX_SIZE_DEFAULT: JmapEmailGetFullMaxSize = JmapEmailGetFullMaxSize(UnsignedInt.liftOrThrow(5))
+  val JMAP_MAX_OBJECT_IN_GET: MaxObjectsInGet = MaxObjectsInGet(UnsignedInt.liftOrThrow(500))
+  val JMAP_MAX_OBJECT_IN_EMAIL_GET_FULL: MaxObjectsInGet = MaxObjectsInGet(UnsignedInt.liftOrThrow(5))
```
Comment (quantranhong1999) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/core/JmapRfc8621Configuration.scala:62]:
```suggestion
```

Likely duplicated with `JMAP_EMAIL_GET_FULL_MAX_SIZE_DEFAULT`.

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/core/JmapRfc8621Configuration.scala [hunk 2]
```diff
@@ -46,6 +46,8 @@ object JmapConfigProperties {
   val DISABLED_CAPABILITIES: String = "disabled.capabilities"
   val JMAP_UPLOAD_QUOTA_LIMIT_PROPERTY: String = "upload.quota.limit"
   val JMAP_EMAIL_GET_FULL_MAX_SIZE_PROPERTY: String = "email.get.full.max.size"
+  val JMAP_GET_MAX_SIZE_PROPERTY: String = "get.max.size"
+  val JMAP_SET_MAX_SIZE_PROPERTY: String = "set.max.size"
```
Comment (quantranhong1999) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/core/JmapRfc8621Configuration.scala:50]: documentation needed

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/Method.scala [hunk 3]
```diff
@@ -56,6 +56,35 @@ trait Method {
 trait WithAccountId {
   def accountId: AccountId
 }
+trait GetRequest extends ValidableRequest {
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/Method.scala:59]: Bit of breathing with a line skip between traits maybe?

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/Method.scala [hunk 4]
```diff
@@ -56,6 +56,35 @@ trait Method {
...
+
+  override def validate(configuration: JmapRfc8621Configuration): Either[Exception, GetRequest] =
+    if (idCount.exists(value => value > configuration.maxObjectsInGet.value.value)) {
+      Left(RequestTooLargeException(s"Too many items in a get request ${this.getClass}. " +
+        s"Got $idCount items instead of maximum ${configuration.maxObjectsInGet.value}."))
+    } else {
+      scala.Right(this)
+    }
+}
+trait SetRequest extends ValidableRequest {
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/Method.scala:70]: idem

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/Method.scala [hunk 5]
```diff
@@ -56,6 +56,35 @@ trait Method {
...
+  def validateIdCounts(configuration: JmapRfc8621Configuration) = {
+    if (idCount > configuration.maxObjectsInGet.value.value) {
+      Left(RequestTooLargeException(s"Too many items in a set request ${this.getClass}. " +
+        s"Got $idCount items instead of maximum ${configuration.maxObjectsInSet.value}."))
+    } else {
+      scala.Right(this)
+    }
+  }
+}
+trait ValidableRequest {
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/Method.scala:85]: idem

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/Method.scala [hunk 6]
```diff
@@ -56,6 +56,35 @@ trait Method {
...
+  def validate(configuration: JmapRfc8621Configuration): Either[Exception, ValidableRequest]
+}
 trait MethodRequiringAccountId[REQUEST <: WithAccountId] extends Method {
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/Method.scala:88]: idem

**Review by Arsnael — APPROVED** [review]

**Review by vttranlina — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

**Pull request merged**